DataFormats/TrackReco/src/TrackBase.cc

```diff
@@ -144,3 +144,20 @@ TrackBase::TrackAlgorithm TrackBase::algoByName(const std::string &name) {
   // cast
   return TrackAlgorithm(index);
 }
+
+double TrackBase::dxyError(Point const &vtx, math::Error<3>::type const &vertexCov) const {
```

Comment:
This is the proposed way to propagate uncertainties properly for the dxy measurement with respect to a given PV. 

Going from the original frame (dxy) to the new frame (dxy') translated from PV coordinates (xv, yv)  it uses the transformation:
dxy' = dxy + x_v*sin(phi) - y_v*cos(phi)

And propagates accordingly using the track and vertex covariance matrices plus gradients

(0,0,xv*cos(phi) + yv*sin(phi), 1, 0) (gradient with respect to the track parameters in the curvilinear basis)

(sin(phi), -cos(phi), 0) (gradient with respect to the vertex parameters)